### ISSUE TITLE
TypeVar expects arguments to be parametrised

### ISSUE THREAD

**Issue by layday** [microsoft/pyright#1077]
**Describe the bug**
Pyright reports that it expects type arguments to be passed to generic types passed to a `TypeVar` but this is invalid and will produce a corresponding 'bound type cannot be generic' error.

**To Reproduce**
Type check the snippet below.

**Expected behavior**
That Pyright would not require generic types to be parametrised when creating `TypeVar`s (although, if that were actually possible, it would be awesome).

**Screenshots or Code**
```python
from typing import Tuple, TypeVar
ListOrTuple = TypeVar('ListOrTuple', list, Tuple)
```

**VS Code extension or command-line**
Extension, 1.1.75

**Comment by erictraut** [comment]
Where are you seeing the "bound type cannot be generic" error? I'm not able to repro this.

This runs fine for me (in Python 3.8):

```python
from typing import Generic, List, Tuple, TypeVar

ListOrTuple = TypeVar("ListOrTuple", List[int], Tuple[int, ...])

def foo(a: ListOrTuple) -> ListOrTuple:
    return a

foo([2, 3, 4])

class Foo(Generic[ListOrTuple]):
    pass

Foo()
```

**Comment by layday** [comment]
Ah, I must've tried to pass a type variable instead of a regular type - I got things mixed up while attempting to get 'high kinded types' to work.  Apologies, you can disregard this.
